specs/eventing/overview.md

```diff
@@ -1,97 +1,88 @@
-[Trigger](spec.md#kind-trigger), [Broker](spec.md#kind-broker),
-[Subscription](spec.md#kind-subscription) and abstract resource definitions for
-[Channels](spec.md#kind-channel).
+The Knative Eventing API provides primitives for two common event-processing
+patterns (credit to James Urquhart for the formulation):
 
-With extensibility and composability as a goal of Knative Eventing, the eventing
-API defines several resources that can be reduced down to well understood
-contracts. These eventing resource interfaces may be fulfilled by other
-Kubernetes objects and then composed in the same way as the concrete objects.
-The interfaces are ([Addressable](interfaces.md#addressable),
-[Callable](interfaces.md#callable)). For more details, see
-[Interface Contracts](interfaces.md).
+* Point-to-point asynchronous communication ([`messaging.knative.dev`](#messaging))
 
-- A **Trigger** describes a filter on event attributes which should be delivered
-  to an _Addressable_.
+* Content-based event routing ([`eventing.knative.dev`](#eventing))
 
-- A **Broker** provides a bucket of events which can be selected by attribute.
+The other two patterns James identifies are log-stream processing and complex
+workflows; these are not currently addressed by Knative Eventing.
 
-<!-- https://drive.google.com/open?id=1CXRvT2g6sxk6-ZrwYcSf2BahCNVlLTLNZkm-laQitMg -->
+In addition to the primitives needed to express the above patterns, Knative
+Eventing defines two [_interface contracts_](#interface-contracts) to allow
+connecting multiple types of Kubernetes objects as event senders and recipients
+to the core primitives.
 
-![Broker Trigger Overview](images/broker-trigger-overview.svg)
+<!-- TODO: add a drawing -->
 
-The above diagram shows a _Broker_ ingesting events and delivering to a
-_Service_ only when the _Trigger_ filter matches.
+## Eventing
 
-- A **Subscription** describes the transformation of an event and optional
-  forwarding of a returned event.
+### Broker
 
-- A **Channel** provides event persistence and fanout of events from a
-  well-known input address to multiple outputs described by _Subscriptions_.
+**Broker** provides a central event-routing hub which exposes a URL address
+which event senders may use to submit events to the router. A Broker may be
+implemented using many different underlying event-forwarding mechanisms; the
+broker provides a small set of common event-delivery configuration options and
+may reference additional implementation-specific configuration options via a
+reference to an external object; the format of the external objects is not
+standardized.
 
-<!-- This image is sourced from https://drive.google.com/open?id=10mmXzDb8S_4_ZG_hcBr7s4HPISyBqcqeJLTXLwkilRc -->
+### Trigger
 
-![Resource Types Overview](images/resource-types-overview.svg)
+**Trigger** defines a filtered delivery option to extract events delivered to a
+**Broker** and route them to an **Addressable** destination. Trigger implements
```

Comment:
Since we haven't used the Addressable yet, wonder if it might be good to include a link to it the first time we introduce it?

Reply:
Combining with your comment below, I think I may introduce the two partial schemas, **Addressable** and **Destination** (I think Destination is probably not a true duck type because it doesn't contain ObjectMeta and so the object can be anywhere in the object, but it's still a useful building block).

I was trying to cut down on the number of concepts, but looking back, Destination does help clarify a lot of cases.

Reply:
I've also moved the interfaces above the concrete types.